community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -201,9 +203,10 @@ final void forceClose()
         }
     }
 
-    final String authSubjectName()
+    final String username()
     {
-        return transaction.mode().name();
+        AccessMode mode = transaction.mode();
```

Comment:
Always, AuthSubject extends AccessMode. This is in need of refactoring and hopefully that hopefully will happen when security support is added to core api. For now, we need this dance and this situation will arise in community.